File: backend/src/main/java/isep/webapp/terasync/service/odSystem/AddressService.java
package isep.webapp.terasync.service.odSystem;

import isep.webapp.terasync.model.odSystem.Address;
import isep.webapp.terasync.model.query.update.ValueByField;
import isep.webapp.terasync.repository.odSystem.AddressRepository;
import isep.webapp.terasync.service.EntityService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class AddressService extends EntityService<Address, AddressRepository> {
    @Autowired
    public AddressService(AddressRepository entityRepository) {
        super(entityRepository);
    }

    @Transactional
    @Override
    public Integer deleteEntityById(Integer id) {
        return entityRepository.deleteEntityById(id);
    }
}

File: front/src/app/service/odSystem/address.service.ts
import {Injectable} from "@angular/core";
import {EntityService} from "../entity.service";
import {HttpClient} from "@angular/common/http";
import {Address} from "../../model/odSystem/address";

@Injectable({
  providedIn: 'root'
})
export class AddressService extends EntityService<Address> {

  constructor(http: HttpClient) {
    super(http, "address");
  }
}
